src/SDK/Metrics/AggregationTemporalitySelector.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Metrics;
+
+use OpenTelemetry\SDK\Metrics\Data\Temporality;
+
+/**
+ * Determine the temporality for a given instrument type, as a function of temporality preference.
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.48.0/specification/metrics/sdk_exporters/otlp.md#additional-environment-variable-configuration
+ * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.48.0/specification/metrics/sdk.md#metricreader
+ *
+ */
+class AggregationTemporalitySelector
+{
+    public static function alwaysCumulative(): AggregationTemporalitySelectorInterface
+    {
+        return new class() implements AggregationTemporalitySelectorInterface {
+            public function temporality(MetricMetadataInterface $metric): ?Temporality
+            {
+                return match ($metric->instrumentType()) {
+                    InstrumentType::GAUGE, InstrumentType::ASYNCHRONOUS_GAUGE => Temporality::DELTA,
```

Comment:
I didn't see this explicitly in spec, but it's based on a suggested change from @Nevay in the last review...

Reply:
My suggestion https://github.com/open-telemetry/opentelemetry-php/pull/1715#discussion_r2349716137 was only regarding synchronous gauge with lowmemory preference.
If we return `DELTA` for a synchronous instrument with delta preference, we must also return `DELTA` for lowmemory preference, otherwise lowmemory would increase memory consumption.

Reply:
I found Java's tests and replicated them, fixing `gauge` temporality to match Java's expected outputs.